x-pack/legacy/plugins/siem/public/pages/detection_engine/signals/actions.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import moment from 'moment';
+
+import { updateSignalStatus } from '../../../containers/detection_engine/signals/api';
+import { SendSignalsToTimelineActionProps, UpdateSignalStatusActionProps } from './types';
+import { TimelineNonEcsData } from '../../../graphql/types';
+
+export const getUpdateSignalsQuery = (eventIds: Readonly<string[]>) => {
+  return {
+    query: {
+      bool: {
+        filter: {
+          terms: {
+            _id: [...eventIds],
+          },
+        },
+      },
+    },
+  };
+};
+
+export const getFilterAndRuleBounds = (
+  data: TimelineNonEcsData[][]
+): [string[], number, number] => {
+  const stringFilter = data?.[0].filter(d => d.field === 'signal.rule.filters')?.[0]?.value ?? [];
```

Comment:
if a string why default to Array

Reply:
I am wondering if we should write a query to get the min and max of the time with an aggregation. It might be cleaner and safer. what do you think? no need to do it now, just wondering

Reply:
Looks like it defaults to Array as a byproduct of `createTimeline()`'s `filters` arg preferring an `[]`. 

And I like that idea -- that also opens possibilities around the ability to send all signals for the current query (`select all signals on all pages`) as well, which we don't currently allow.